### PR TITLE
correct ordered factor definition

### PR DESCRIPTION
**Pull request by leorjorge**
Following discussion in #1122 , clarifies that ordered factors do not behave like numbers.

### CODE REVIEW

**Comment by leorjorge** [comment]
To clarify, this PR attempts to correct the obvious errors: the definition and the typo. 

I kept the more opinionated part discouraging people from using ordered factors, even if I don't really agree with it and they are quite important at least in the social sciences (and ecology to a lesser extent). In my experience with ecological data, they're used less often than they should, with people just treating these as either numeric variables or regular factors to avoid the use of more complicated models.

File: factors.qmd [hunk 1]
```diff
@@ -421,9 +421,9 @@ In practice, `ordered()` factors behave very similarly to regular factors.
...
--   If you use an ordered function in a linear model, it will use "polygonal contrasts". These are mildly useful, but you are unlikely to have heard of them unless you have a PhD in Statistics, and even then you probably don't routinely interpret them. If you want to learn more, we recommend `vignette("contrasts", package = "faux")` by Lisa DeBruine.
+-   If you use an ordered predictor in a linear model, it will use "polynomial contrasts". These are mildly useful, but you are unlikely to have heard of them unless you have a PhD in Statistics, and even then you probably don't routinely interpret them. If you want to learn more, we recommend `vignette("contrasts", package = "faux")` by Lisa DeBruine.
 
-Given the arguable utility of these differences, we don't generally recommend using ordered factors.
+Ordered categories are frequently used in the social sciences but not so common in other fields, so unless you are faced with a dataset that has clearly ranked categories we don't generally recommend using ordered factors.
```
Comment (hadley) [inline factors.qmd:426]: What's the impact on analysis of using them? Is it that the social sciences are using packages that have more behaviour differences between factors and ordered factors?

Comment (leorjorge) [reply factors.qmd:426]: Yeah, my understanding is that as social sciences have so many survey-based data in a ranked scale, they use more often ordinal regression and other similar methods to deal with ordered categories both as predictors and as responses. For response variables, there is quite a bit of literature on ordinal regressions. these usually employ cumulative logit distributions. Methods to deal with ordered factors as predictors in a better way than these polynomial contrasts do seem to be much more recent, but are getting increasing use, at least in Bayesian analyses (se https://bpspsychub.onlinelibrary.wiley.com/doi/10.1111/bmsp.12195 for example).

**Comment by hadley** [comment]
Thanks!

**Pull request merged**